network.go

```diff
@@ -135,3 +152,34 @@ func getIPFromHeader(r *http.Request, name string) net.IP {
 
 	return nil
 }
+
+func extractSubFromJWT(jwtString, secret string) (string, error) {
+	claims := &jwt.StandardClaims{}
+	token, err := jwt.ParseWithClaims(jwtString, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(secret), nil
+	})
+	if err != nil {
+		return "", err
+	}
+	if !token.Valid {
+		return "", http.ErrLineTooLong
```

Comment:
Maybe you should create a sentinel error here inside this file to express that the JWT _(token)_ is invalid.
Same goes if the JWT was not found.